cluster/minio.go

```diff
@@ -112,3 +112,19 @@ func RemoveMinioUser(sgt *StorageGroupTenant, tenantConf *TenantConfiguration, u
 func tagErrorAsMinio(err error) error {
 	return fmt.Errorf("MinIO: %s", err.Error())
 }
+
+// minioIsReady determines whether the MinIO for a tenant is ready or not
+func minioIsReady(ctx *Context) (bool, error) {
+	// Get tenant specific MinIO client
+	minioClient, err := newTenantMinioClient(ctx, ctx.Tenant.ShortName)
```

Comment:
You should just use /minio/healthcheck/live 

```
func testLivenessEndpoint(endpoint string) {
	startTime := time.Now()
	function := "testLivenessEndpoint"

	u, err := url.Parse(fmt.Sprintf("%s%s", endpoint, livenessPath))
	if err != nil {
		// Could not parse URL successfully
		failureLog(function, nil, startTime, "", "URL Parsing for Healthcheck Liveness handler failed", err).Fatal()
	}

	tr := &http.Transport{
		TLSClientConfig: &tls.Config{InsecureSkipVerify: u.Scheme == "https"},
	}
	client := &http.Client{Transport: tr, Timeout: timeout}
	resp, err := client.Get(u.String())
	if err != nil {
		// GET request errored
		failureLog(function, nil, startTime, "", "GET request failed", err).Fatal()
	}
	if resp.StatusCode != http.StatusOK {
		// Status not 200 OK
		failureLog(function, nil, startTime, "", fmt.Sprintf("GET /minio/health/live returned %s", resp.Status), err).Fatal()
	}

	defer resp.Body.Close()
	defer successLogger(function, nil, startTime).Info()
}
```

It doesn't need an authenticated call - also doing ListBuckets() might get expensive.

Reply:
I think liveness is not going to be enough, we need MinIO to be ready to accept new users or configuration changes, if liveness is reported but we cannot make `mc admin user add` then it's not good, we need MinIO to be ready to take regular requests, not sure if this is what readiness reports on

If we have a 4 node cluster, how many nodes do we need online before we can add a user or change a configuration?

Reply:
Ready, using Head Bucket